documentation/operators/src/testing/node-api-check.md

```diff
@@ -0,0 +1,118 @@
+# Node API Check
+
+> Any syntax in `<>` brackets is a user's unique variable/version. Exchange with a corresponding name without the `<>` brackets.
+
+Operating a `nym-node` is not a *"set and forget"* endeavor, it takes some work. To diagnose node performance querying APIs is a good knowledge to have. There are two main places to look for API endpoints regarding `nym-node`:
+
+- [`openapi.json`](https://validator.nymtech.net/api/v1/openapi.json): a list of all endpoints
+- [Swagger UI page](https://validator.nymtech.net/api/swagger/index.html)
+
+Besides that, Gateway operators can check out their node performance, connectivity and much more on [harbourmaster.nymtech.net](https://harbourmaster.nymtech.net/).
+
+### Basic API usage
+
+For information about available endpoints and their status, you can refer to:
+```
+# for http
+http://<IP>:8080/api/v1/swagger/#/
+
+# for https reversed proxy
+https://<DOMAIN>/api/v1/swagger/#/
+```
+
+For example to determine which mode your node is running, you can check `:8080/api/v1/roles` endpoint:
+```
+# for http
+http://<IP_ADDRESS>:8080/api/v1/roles
+
+# for https reversed proxy
+https://<DOMAIN>/api/v1/roles
+```
+
+## `node_api_check.py`
+
+To make this a bit easier, we made a CLI tool quering all vailable API endpoints based on node `Identity Key` (further denoted as `<ID_KEY>`) called `node_api_check.py`. To diagnose your node performance, whether by yourself or by sharing an output in our [operator channel](https://matrix.to/#/#operators:nymtech.chat), this tool provides you with a quick overview of live data. We recommend to run this checker alongside [`nym_gateway_probe`](gateway-probe.md) to triage both performance and an actual routing.
+
+Besides querying any bonded node APIs, `nym_api_check` has a function counting all existing nodes in provided version.
+
+### Setup
+
+#### Pre-requsities
+
+**Python3**
+
+1. Start with installing Python3:
+```sh
+sudo apt install Python3
```

Comment:
pedantic, but if someone were to copy this line it will fail due to the capital P

Reply:
oups, good point